### ISSUE TITLE
Allow use of `as` and `is` on arbitrary expressions, not just the parameter

### ISSUE THREAD

**Issue by zspitz** [zzzprojects/System.Linq.Dynamic.Core#457]
Currently, `as` is defined as taking only one argument: the type name; it can only be applied to the parameter. This:
```csharp
class Foo {}
class Bar : Foo {}
class Baz {
    public Foo Foo1 {get;set;} = new Bar();
}

var selector = $"as(\"{typeof(Bar).FullName}\")";
```
is defined as the equivalent of:
```csharp
x => x as Bar;
```
But there is no way to express the equivalent of the following:
```csharp
Expression<Func<Baz, Bar>> expr = x => x.Foo1 as Bar;
```
Either of the following:
``` csharp
var selector = $"as(Foo1, \"{typeof(Bar).FullName}\")";
var selector = $"as(\"{typeof(Bar).FullName}\", Foo1)";
```
fails with:

> **System.Linq.Dynamic.Core.Exceptions.ParseException:** 'The 'as' function requires one argument'

---

The same thing applies to `is`. Either of the following:
``` csharp
var selector = $"is(Foo1, \"{typeof(Bar).FullName}\")";
var selector = $"is(\"{typeof(Bar).FullName}\", Foo1)";
```
fails with the above error.

**Comment by StefH** [comment]
Hello @zspitz ,

For this moment it needs to be a constant value (Type or string).

This question is also related #459.

Closing this issue.

**Comment by zspitz** [comment]
This has now been fixed with #567, see #566.

**Comment by zspitz** [comment]
> For this moment it needs to be a constant value (Type or string).

@StefH Note that this issue is unrelated to passing the type/string as a parameter object to the Dynamic Linq expression; this issue is about applying `as`/`is` to an arbitrary expression.
